nncf/common/quantization/quantizer_propagation/solver.py

```diff
@@ -257,6 +262,10 @@ def __init__(self,
           trainable quantizer parameters.
         :param run_consistency_checks: Whether to run internal consistency checks at each propagataion step.
         :param quantize_outputs: Whether to insert additional quantizers right before each of the model outputs.
+        :param post_processing_metatypes: The framework specific TopK and NonMaximumSuppression NNCF Metatypes
+         that are used for automatic ignoring post-processing nodes, connected with TopK and NonMaximumSuppression.
+         The first element in the list should be TopK metatype, the second - NonMaximumSuppression.
+         If None automatic ignoring will be skipped.
```

Comment:
The description does not correspond to the logic. You are describing the specific case here while the logic is actually agnostic of the actual meaning of metatypes provided in this parameter. 
`The first element in the list should be TopK metatype, the second - NonMaximumSuppression.` - we'll have to rework this.

Reply:
Actually, the logic of adding the nodes to the ignored behaves in two different ways depending on the visited metatypes. This logic is described in docstring:
```
If the path with the nodes has the node with NonMaxSuppressionMetatype,
all the nodes in this path will be added into ignored.
If the path with the nodes has the node with TopKMetatype,
all the nodes starting from TopKMetatype in this path will be added into ignored.
```

But you gave a good idea, with the new logic with forward traversing till any weight node, it could be replaced with a simpler one - If during backward traversing the algorithm faces any node from post-processing list then add all traversed nodes to ignored.